routes/sessions.js

```diff
@@ -12,56 +12,61 @@ module.exports = function (app, opts) {
     new AllowedUsersFinder(request.body.renderingId, opts)
       .perform()
       .then(function (allowedUsers) {
+        if (!opts.authSecret) {
+          throw new Error('Your Forest authSecret seems to be missing. Can ' +
+            'you check that you properly set a Forest authSecret in the ' +
+            'Forest initializer?');
+        }
+
         if (allowedUsers.length === 0) {
-          return response.status(401).send({
-            errors: [{
-              detail: 'Forest cannot retrieve any users for the project ' +
-                'you\'re trying to unlock.'
-            }]
-          });
+          throw new Error('Forest cannot retrieve any users for the project ' +
+            'you\'re trying to unlock.');
         }
 
         var user = _.find(allowedUsers, function (allowedUser) {
-          return allowedUser.email === request.body.email &&
-            bcrypt.compareSync(request.body.password, allowedUser.password);
```

Comment:
I think the real optimization is here.
Async bcrypt uses another thread to compute the comparison and doesn't lock the event loop, which is nice but doesn't speed up the process for 1 user